dnadoh/synthesize.py

```diff
@@ -0,0 +1,335 @@
+
+    def make_gsex(self, person, reference, individual):
+        """Generate a random genetic sex.
+
+        Genetic sexes are chosen using weighted probabilities.
+        """
+        person.gsex = random.choices(self.GSEX, weights=self.GSEX_PROBS)[0]
+
+    def make_weight(self, person, reference, individual):
+        """Generate a random weight.
+
+        Weights are chosen from a normal distribution based on genetic sex.
+        """
+        assert person.gsex is not None
+        mean = self.WEIGHT_MEANS[person.gsex]
+        std_dev = mean * self.WEIGHT_RSD
+        person.weight = _truncate(random.gauss(mean, std_dev), 1)
+
+
+def adjust_all(genomes, people, func):
+    """Adjust a single value based on an SNP.
+
+    1.  Choose one of the mutation locations at random.
+    2.  Select a base other than the one in the reference genome at that location.
+    3.  Apply the mutation function to everyone who has that base at that location.
+    """
+    if not genomes.locations:
+        return
+    loc = random.choice(genomes.locations)
+    candidates = _other_bases(genomes.reference, loc)
+    marker = random.choice(list(candidates))
+    for person in people:
+        if person.genome[loc] == marker:
+            func(person)
+
+
+def adjust_weight(person):
+    """Adjust a person's weight upward by 10%."""
+    person.weight = _truncate(1.1 * person.weight, 1)
+
+
+# --------------------------------------------------------------------------------------
+# I/O
+# --------------------------------------------------------------------------------------
+
+
+def _write(output_stem, genomes, people):
+    """Write people as CSV."""
+    assert people
+    _write_overall(output_stem, people)
+    _write_reference_genome(output_stem, genomes)
+    _write_variants(output_stem, genomes, people)
+    _write_people(output_stem, people)
+
+
+def _write_overall(output_stem, people):
+    """Write DNA sequences and people for reference."""
+    filename = f"{output_stem}-overall.csv"
+    headings = people[0].dict().keys()
+    with open(filename, "w") as raw:
+        writer = csv.DictWriter(raw, fieldnames=headings)
+        writer.writeheader()
+        for person in people:
+            writer.writerow(person.dict())
+
+
+def _write_reference_genome(output_stem, genomes):
+    """Save the reference genome and related information."""
+    filename = f"{output_stem}-reference.json"
+    data = {"genome": genomes.reference, "locations": list(sorted(genomes.locations))}
+    with open(filename, "w") as writer:
+        json.dump(data, writer)
+
+
+def _write_variants(output_stem, genomes, people):
+    """Write one variant file per person."""
+    width = len(str(len(people)))
+    for (pid, person) in enumerate(people):
+        pid_str = str(pid).zfill(width)
+        filename = f"{output_stem}-pid{pid_str}.csv"
+        with open(filename, "w") as raw:
+            writer = csv.DictWriter(raw, fieldnames=["loc", "base"])
```

Comment:
loc is based on the index, right? so 0-based coordinate? 
if so , it's more conventional to have the location as 1-based. (i.e., a loc of 0 wouldn't exist)

Reply:
yes and sure - easy to change.